docs/data/Allas/using_allas/python_boto3.md

```diff
@@ -1,57 +1,102 @@
+for example
+[AWS SDK for Pandas](https://aws-sdk-pandas.readthedocs.io/en/stable/)
+and
+[GDAL-based Python libraries](https://github.com/csc-training/geocomputing/blob/master/python/allas/working_with_allas_from_Python_S3.py)
+for spatial data analysis.
+
+!!! note ""
+    Remember to avoid handling the same objects with both S3 and SWIFT, as
+    [they work differently with large objects](../introduction.md#protocols).
 
 ## Installation
 
-`boto3` is available for Python 3.8 and higher and can be [installed](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#installation) with `pip` or `conda`.
+`boto3` is available for Python 3.8 and higher.
+It can be
+[installed on a personal device](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#installation)
+with `pip` or `conda`.
 
-```
+```bash
+# pip
 pip install boto3
+
+# conda
+conda install boto3
 ```
 
-### `boto3` in CSC supercomputers
-Some existing [Python modules](../../../apps/python.md#pre-installed-python-environments) might have `boto3` pre-installed, for example [geoconda](../../../apps/geoconda.md). 
-To other modules, it is possible to add `boto3` with [pip](../../../support/tutorials/python-usage-guide.md#installing-python-packages-to-existing-modules).
+### `boto3` on CSC supercomputers
 
+The pre-existing
+[`geoconda`](../../../apps/geoconda.md) and
+[`biopythontools`](../../../apps/biopython.md)
+Python modules have `boto3` installed. 
+It is also possible to
+[add `boto3` on top of other modules](../../../support/tutorials/python-usage-guide.md#installing-python-packages-to-existing-modules)
+using `pip`.
 
 ## Configuring S3 credentials
 
-If you have not used Allas with S3 before, then first [create S3 credentials](s3_client.md#getting-started-with-s3cmd). The credentials are saved to `~/.aws/credentials` file, so they need to be set only once from a new computer or when changing project. The credential file can be also copied from one computer to another.
-
-In CSC supercomptuers [`allas` module](s3_client.md#configuring-s3-connection-in-supercomputers) can be used with `allas-conf --mode s3cmd` to configure the credentials.
+The easiest way to set up S3 credentials for using `boto3` is by
+[configuring an S3 connection on a CSC supercomputer](s3_client.md#configuring-s3-connection-in-supercomputers).
+After running `allas-conf --mode s3cmd`, the credentials are stored on the
+supercomputer in `~/.aws/credentials`, where `boto3` looks for them
+automatically.
+
+If you are accessing Allas from a personal workstation,
+you can copy `.aws` and its contents to your workstation home directory
+[using a file transfer tool](../../moving/index.md), e.g. by running
+`scp -r <username>@puhti.csc.fi:~/.aws $HOME`. As above, `boto3` automatically
+looks for your credentials in your local home directory.
+
+Note that only one set of credentials can be stored in `~/.aws/credentials`.
```

Comment:
This is not true, see https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html and profile_name. I have several profiles in my credentials file from earlier. 
(On a conference today and then for 4 weeks on holidays, so can not properly review this.)

Reply:
Thank you for pointing that out! I will make the appropriate changes.

Reply:
That might be true. You can ask also Robin or Henrik, I would recall that Puhti/Mahti/LUMi web interface Allas connection made credential per project, might remember wrong.

Reply:
Thanks! That works :)